internal/middleware/enforce_identity.go

```diff
@@ -0,0 +1,89 @@
+package middleware
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+	"github.com/redhatinsights/platform-go-middlewares/identity"
+)
+
+var (
+	InvalidOrMissingOrgErr = errors.New("x-rh-identity header has an invalid or missing org_id")
+	MissingTypeErr         = errors.New("x-rh-identity header is missing type")
+)
+
+func doError(ctx context.Context, w http.ResponseWriter, code int, reason string) {
+	logger := ctxval.Logger(ctx)
+	logger.Warn().Msgf("Failed to enforce the Identity header: %s", reason)
+	http.Error(w, reason, code)
+}
+
+// EnforceIdentity extracts the X-Rh-Identity header and places the contents into the
+// request context.  If the Identity is invalid, the request will be aborted.
+func EnforceIdentity(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		logger := ctxval.Logger(r.Context())
+		rawHeaders := r.Header["X-Rh-Identity"]
```

Comment:
Can you please change all log lines in this file to read "X-Rh-Identity"? Because this is a map, it must be capitalized, yet we log "x-rh-identity" which is an invalid value I believe.

Reply:
There is one more.... gosh. Ok. Fine